File: android_fragment/app/src/main/java/com/support/generator/module/main/MainActivitySubComponent.java
package com.support.generator.module.main;

import android.app.Activity;

import dagger.Binds;
import dagger.Subcomponent;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

/**
 * @author: leellun
 * @data: 2018/12/24.
 */
@Subcomponent(modules = {MainActivityModule.class})
public interface MainActivitySubComponent extends AndroidInjector<MainActivity> {

  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<MainActivity> {
  }

  //TODO REMINDER: move the line below to AppModule's @Module(...) annotation
  //MainActivitySubComponent.class
  //TODO REMINDER: move this to ActivitiesInjectorFactories module

}

File: example/app/src/main/java/gauges/mobile/github/com/mvpapplication/module/main/MainActivitySubComponent.kt
package gauges.mobile.github.com.mvpapplication.module.main

import dagger.Subcomponent
import dagger.android.AndroidInjector

/**
 * @author: leellun
 * @data: 2018/12/19.
 *
 */
@Subcomponent(modules = arrayOf(MainActivityModule::class))
interface MainActivitySubComponent : AndroidInjector<MainActivity> {

    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<MainActivity>()

    //TODO REMINDER: move the line below to AppModule's @Module(...) annotation
    //MainActivitySubComponent::class

    //TODO REMINDER: move this to ActivitiesInjectorFactories module
}